src/golang/cmd/migrator/versions/000019_add_serialization_value_to_param_op/utils.go

```diff
@@ -0,0 +1,115 @@
+package _000019_add_serialization_value_to_param_op
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/utils"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	serilization_type_key   = "serialization_type"
+	value_key               = "val"
+	spec_field              = "spec"
+	pythonExecutorPackage   = "aqueduct_executor"
+	typeInferencePythonPath = "migrators.parameter_val_type_inference_000019.main"
+)
+
+type Operator struct {
+	Id     uuid.UUID `db:"id" json:"id"`
+	OpSpec Spec      `db:"spec" json:"spec"`
+}
+
+func getAllOperators(
+	ctx context.Context,
+	db database.Database,
+) ([]Operator, error) {
+	query := "SELECT id, spec FROM operator;"
+
+	var response []Operator
+	err := db.Query(ctx, &response, query)
+	return response, err
+}
+
+func updateParamOperatorWithNewSpec(
+	ctx context.Context,
+	operator Operator,
+	db database.Database,
+) error {
+	// If the serialization_type is present we should consider this already migrated
+	if _, exists := operator.OpSpec.Param[serilization_type_key]; exists {
+		return nil
+	}
+
+	param_val := operator.OpSpec.Param[value_key]
+	// Launch the Python job to infer the type of the parameter value
+	cmd := exec.Command(
+		"python3",
+		"-m",
+		fmt.Sprintf("%s.%s", pythonExecutorPackage, typeInferencePythonPath),
+		"--val",
+		param_val,
+	)
+	cmd.Env = os.Environ()
+
+	var outb, errb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &errb
+
+	err := cmd.Run()
+	if err != nil {
+		log.Errorf("Error running Python migration job. Stdout: %s, Stderr: %s.", outb.String(), errb.String())
+		return err
+	}
+
+	fmt.Println(fmt.Sprintf("WE GOT %s", outb.String()))
+	outputs := strings.Split(outb.String(), "\n")
+	param_type := outputs[0]
+	param_val = outputs[1]
+	operator.OpSpec.Param[serilization_type_key] = param_type
+
+	// We also change the param value to be a base64 encoding
+	operator.OpSpec.Param[value_key] = param_val[2:len(param_val)-1]
```

Comment:
@HarisChoudhary could you make a comment why we are doing `[2:len(param_val)-1]`? In general I think it's cleaner to have the Python side write the output to a file, which is then read by the Golang side (like in migration 16) instead of trying to parse the output to stdout. But if this is too much work and the current way works, I'm fine with keeping it as is for this time.

Reply:
@cw75 I didnt mean for this to get reviews yet despite the push, my other implementation doesnt have this part